### PR TITLE
Wip update solution

### PR DESCRIPTION
**Pull request by realAndrewCohn**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,11 @@
-import React, { Component } from 'react';
-import PokemonPage from "./components/PokemonPage";
-import './App.css';
+import React from 'react'
+import PokemonIndex from './components/PokemonIndex'
```
Comment (realAndrewCohn) [inline src/App.js:2]: renamed pokemonPage to pokemonIndex

File: src/App.js [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
-class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <PokemonPage />
-      </div>
-    );
-  }
-}
+const App = () => (
```
Comment (realAndrewCohn) [inline src/App.js:5]: made this a functional component

File: src/App.test.js [hunk 3]
```diff
@@ -1,9 +0,0 @@
-import React from 'react';
```
Comment (realAndrewCohn) [inline src/App.test.js:1]: if we aren't doing tests, we can safely delete this file

File: src/components/PokemonCard.js [hunk 4]
```diff
@@ -1,35 +1,29 @@
...
-              {name}
-            </div>
-          </div>
-          <div className="extra content">
-            <span>
-              <i className="icon heartbeat red" />
-              {hp} hp
-            </span>
-          </div>
+const PokemonCard = ({ pokemon, toggleImage }) => {
```
Comment (realAndrewCohn) [inline src/components/PokemonCard.js:4]: made this one functional too

File: src/components/PokemonCollection.js [hunk 5]
```diff
@@ -1,19 +1,13 @@
...
-				{this.props.pokemon.map(
-					pokemon => {
-						return <PokemonCard key={pokemon.id} pokemon={pokemon} toggleImage={this.props.toggleImage} />
-					}
-				)}
-			</Card.Group>
-		)
-	}
-};
+const PokemonCollection = ({ pokemon, toggleImage }) => (
```
Comment (realAndrewCohn) [inline src/components/PokemonCollection.js:5]: functional as well

File: src/components/PokemonIndex.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  toggleImage = pokemon => {
+    const col = this.state.pokemonCollection
+    const i = col.indexOf(pokemon)
+    this.setState({
+      pokemonCollection: [
+        ...col.slice(0, i),
+        // initially pokemon.isClickes is undefined; inverting that falsey value makes it true
+        { ...pokemon, isClicked: !pokemon.isClicked },
```
Comment (realAndrewCohn) [inline src/components/PokemonIndex.js:31]: i think isClicked is a little clearer than `clicked`

File: src/components/PokemonIndex.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      .then(res => res.json())
+      .then(pokemonCollection => this.setState({ pokemonCollection: pokemonCollection }))
+      .catch(e => console.error(e))
+  }
+
+  handleSearchChange = (e, { value }) => {
+    this.setState({ searchTerm: value })
+  }
+
+  toggleImage = pokemon => {
```
Comment (realAndrewCohn) [inline src/components/PokemonIndex.js:24]: @morgvanny i'm still not sure it makes sense to move this up to pokemonIndex. Why do we not want this in PokemonCard?

File: src/components/PokemonPage.js [hunk 8]
```diff
@@ -1,54 +0,0 @@
-import React from "react";
```
Comment (realAndrewCohn) [inline src/components/PokemonPage.js:1]: this file was renamed

File: src/logo.svg [hunk 9]
```diff
@@ -1,7 +0,0 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3">
```
Comment (realAndrewCohn) [inline src/logo.svg:1]: not using this so it's gone now

**Comment by realAndrewCohn** [comment]
also, this PR addresses #1 and #2

**Review by morgvanny — APPROVED** [review]
seemsGood

**Pull request merged**